File: ExpendTextView/app/src/main/java/com/cyq/expendtextview/MainActivity.java
package com.cyq.expendtextview;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.cyq.expendtextview.adapter.MyAdapter;
import com.cyq.expendtextview.bean.TestBean;
import com.cyq.expendtextview.view.Utils;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    private RecyclerView mRecyclerView;
    private MyAdapter mAdapter;
    private List<TestBean> mList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mRecyclerView = findViewById(R.id.recyclerView);
        int width = getWindowManager().getDefaultDisplay().getWidth() - Utils.dp2px(this, 20);
        //构造假数据
        for (int i = 0; i < 30; i++) {
            TestBean bean = new TestBean();
            if (i % 3 == 0) {
                bean.setStr("测试一行文字的");
            } else if (i % 4 == 0) {
                bean.setStr("测试二行文字的，测试二行文字的，测试二试二行文字的，测试二行文字的");
            } else if (i % 5 == 0) {
                bean.setStr("测试三行文字的，测试三行文字的，测试三行文字的，测试三行文字的，测试三行文字的，测试三行文字的" +
                        "测试三行文字的测试三行文字的" +
                        "测试三行文字的测试三行文字的测试三行文字的测试三行文字的测试三行文字的测试三行文字的测试三行文字的");
            } else {
                bean.setStr(getResources().getString(R.string.text_content));

            }
            bean.isChecked = false;
            mList.add(bean);
        }
        mAdapter = new MyAdapter(this, mList, width);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        mRecyclerView.setAdapter(mAdapter);
    }
}

File: NestedScroll/app/src/main/java/com/chenyangqi/nested/scroll/coordinator/MainActivity.kt
package com.chenyangqi.nested.scroll.coordinator

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.chenyangqi.nested.scroll.R

class MainActivity : AppCompatActivity() {
    private lateinit var mRecyclerView: RecyclerView
    private lateinit var mAdapter: MyAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val mData: MutableList<String> = ArrayList()
        for (i in 1..10000) {
            mData.add("item = $i")
        }

        mRecyclerView = findViewById(R.id.mRecyclerView)
        mAdapter = MyAdapter(mData)
        mRecyclerView.adapter = mAdapter
        mRecyclerView.layoutManager = LinearLayoutManager(this)
    }

    inner class MyAdapter(private val data: MutableList<String>) :
        RecyclerView.Adapter<MyViewHolder>() {
        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
            val view = LayoutInflater.from(this@MainActivity)
                .inflate(R.layout.item_nested_list, parent, false)
            return MyViewHolder(view)
        }

        override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
            holder.tvItemContent.text = data[position]
        }

        override fun getItemCount(): Int {
            return data.size
        }
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvItemContent: TextView = itemView.findViewById(R.id.tvItemContent)
    }
}